### ISSUE TITLE
Should prefer_final_parameters fire on a super-parameter?

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#3211]
The `prefer_final_parameters` rule has tests indicating that it _should_ report constructor parameters:

```dart
class C {
  C.badValue(String value): this.value = value; // LINT
}
```

but _not_ for field formal parameters:

```dart
class C {
  C.goodValue(this.value); // OK
}
```

So, what about super-parameters? Whether it should or shouldn't, there should be a test.

**Comment by eernstg** [comment]
A super parameter is final even without the `final` modifier, just like an initializing formal, so it seems superfluous to ask for the modifier.

**Comment by srawlins** [comment]
I'm happy to submit the test case.

**Comment by pq** [comment]
Thanks @srawlins!
